### PR TITLE
Change v0.3.0 to mitigate risks on search_path vulnerability

### PR DESCRIPTION
**Pull request by thesocialdev**
BUG: https://phabricator.wikimedia.org/T190240

### CODE REVIEW

**Review by pnorman — CHANGES_REQUESTED** [review]
This doesn't mitigate the CVE. The CVE is about search_path problems in pg_dump and the CVE doesn't impact any software except that which comes from PostgreSQL. What we have is the same mistake as the PostgreSQL client software developers made.

You made this PR against the `search_path_v0.3.0` branch, normally you'd do it against master. Also, shouldn't this PR be against upstream?

**Comment by mdholloway** [comment]
(For the record, it was me who suggested doing a fix against a kartotherian fork first, then upstreaming.)

**Comment by pnorman** [comment]
> (For the record, it was me who suggested doing a fix against a kartotherian fork first, then upstreaming.)

I'd only go for our fork if upstream is unresponsive. While it's quite possible they will be, let's give them a chance.

**Comment by mdholloway** [comment]
So, I have to apologize here if I've been giving bad guidance because I had assumed that mitigating this in client code was simply a matter of making implicit `pg_catalog` references explicit.  It seems there is more needed.  Maybe we need to set the `search_path` here (something loosely along the lines of https://github.com/zalando/spilo/pull/235/files)?  @pnorman what would you suggest is the right thing to do here?

**Comment by thesocialdev** [comment]
Sorry about the long reading.

> This doesn't mitigate the CVE. The CVE is about search_path problems in pg_dump and the CVE doesn't impact any software except that which comes from PostgreSQL. What we have is the same mistake as the PostgreSQL client software developers made.

Changed the description and title, sorry about that. I think I misinterpreted the [documentation](https://wiki.postgresql.org/wiki/A_Guide_to_CVE-2018-1058:_Protect_Your_Search_Path#Overview) at this paragraph:
> The purpose of the release was to address CVE-2018-1058, which describes how a user can create like-named objects in different schemas that can change the behavior of other users' queries and cause unexpected or malicious behavior, also known as a "trojan-horse" attack. Most of this release centered around added documentation that described the issue and how to take steps to mitigate the impact on PostgreSQL databases.

> You made this PR against the `search_path_v0.3.0` branch, normally you'd do it against master. Also, shouldn't this PR be against upstream?

The problem I found with the PR against master is that `v0.3.0` is 38 commits behind and the latest version is incompatible backward. That is why I suggested we fork and create `v0.3.1` temporarily as a fallback if something goes wrong. My solution would be tag `v0.3.1` on the new branch created without changing `master`. What do you think we should do about it?

I agree that would be better to make a PR to upstream [I just started this process](https://github.com/mapbox/postgis-vt-util/pull/21) with the latest version, this way we would have the most recent code and contribute with the official maintainer, but this will lead to some minor refactoring.

Michael suggested that we should go first with the fork and mitigate the risk on the current version used by `osm-bright.tm2source`. I agreed and I think that this way we can fulfill the task's goal with minor changes on the code, but I also think that this is temporary and we shouldn't stop here and continue with the upgrade of the library and proper refactoring in the future.

To summarize, I am doing the following steps:
- Fork v0.3.0 and fix search_path vulnerabilities on the old version in the fork
- Send a PR to fix the latest version in upstream
- Upgrade library at `osm-bright.tm2source`

What do you think about the above strategy?

**Comment by mdholloway** [comment]
See also https://github.com/mapbox/postgis-vt-util/pull/21

@stephanebisson, do you have any time to weigh in on this?

**Comment by pnorman** [comment]
> The problem I found with the PR against master is that v0.3.0 is 38 commits behind and the latest version is incompatible backward.

Just to note, in general my practice for backports is to create a branch in the upstream repo, e.g. v0.3.x and then push to my repo a feature branch, then do a PR to merge the feature branch into the backport branch.
